### ISSUE TITLE
Need a Visual Change in Select Feature by Value

### ISSUE THREAD

**Issue by asgharibraheembaloch** [qgis/QGIS#42670]
it is very difficult for me sometime to track a field name and it's input box when there are more fields, i want a visual change with line or something indicating field name and its input box

![image](https://user-images.githubusercontent.com/46741864/113660799-56413a00-96be-11eb-9da8-64d491425ebe.png)

**Comment by nyalldawson** [comment]
Which QGIS version is this? The dialog looks quite different in current versions...

**Comment by asgharibraheembaloch** [comment]
QGIS 3.8.2 Zanzibar

**Comment by nyalldawson** [comment]
Can you please test on a newer release?

**Comment by asgharibraheembaloch** [comment]
ok i have QGIS 3.14 Pi is this ok or i have to install 3.18 latest

**Comment by nyalldawson** [comment]
Is the issue resolved on 3.14?

**Comment by asgharibraheembaloch** [comment]
i have found something explained in two steps
step 1: when i go into Layer properties>Attribute form>Autogenerate
it open like this
![image](https://user-images.githubusercontent.com/46741864/113668234-5e53a680-96cb-11eb-8fb8-3661a3cb8da6.png)

step 2: when i go into Layer properties>Attribute form>Drag and Drop Designer
it open like this
![image](https://user-images.githubusercontent.com/46741864/113668402-9d81f780-96cb-11eb-8589-ee6b5e2495a2.png)

**Comment by asgharibraheembaloch** [comment]
I have also checked on latest QGIS 3.18.1 Zurich same issue persist

**Comment by gioman** [comment]
> Step 2: when I go into Layer properties>Attribute form>Drag and Drop Designer
> it open like this

@asgharibraheembaloch yes, but as soon as you group the attributes in a tab (as is expected when using the drag and drop designer) you gets what you expects.

**Comment by asgharibraheembaloch** [comment]
@gioman but I think something has to be done so that UI experience is not effected, user doesn't think from developer perspective he works on UI and develop his understanding on tools based on how easy to use it

**Comment by gioman** [comment]
> but I think something has to be done so that UI experience is not effected, user doesn't think from developer perspective he works on UI and develop his understanding on tools based on how easy to use it

@asgharibraheembaloch organizing the fields in tabs using the D&D designer is not something a developer does, is something the user does when this mode is used. Is not supposed to use the D&D designer of then you are not organizing the fiels in tabs and groups.

**Comment by asgharibraheembaloch** [comment]
@gioman UI is changed that is issue, makes difficult for me or someone to understand, don't know what happened at backend, that's why I say something need to change so that understanding is easy

**Comment by gioman** [comment]
> UI is changed that is issue

@asgharibraheembaloch the user is not supposed to use the D&D designer if then is not grouping/tabbing the fields. If he/she does it there is no issue.

**Comment by asgharibraheembaloch** [comment]
@gioman but sometime there is need when I have to work on less fields and I don't want to use so many fields, and I want my form to be concise I think that's why QGIS also added this feature if there is no need than QGIS have to remove this feature

**Comment by gioman** [comment]
> to work on less fields and I don't want to use so many fields

@asgharibraheembaloch sure, but you should tab/group them anyway.

**Comment by asgharibraheembaloch** [comment]
> > UI is changed that is issue
> 
> @asgharibraheembaloch the user is not supposed to use the D&D designer if then is not grouping/tabbing the fields. If he/she does it there is no issue.

in my organization there are fields which are for the database team and when digitizer digitize map he has no concern with db team fields then he want his concerning fields

**Comment by gioman** [comment]
> in my organization there are fields which are for the database team and when digitizer digitize map he has no concern with db team fields then he want his concerning fields

@asgharibraheembaloch at a project/table level field can be hidden without the need to use the d&d designer.

**Comment by asgharibraheembaloch** [comment]
> > in my organization there are fields which are for the database team and when digitizer digitize map he has no concern with db team fields then he want his concerning fields
> 
> @asgharibraheembaloch at a project/table level field can be hidden without the need to use the d&d designer.

@gioman digitizer needs a work on canvas level for example take selection on the canvas and then remove this feature based on that field that's why he use 'Select Feature by Value' i know digitizer work other way but that's the easiest way

**Comment by uclaros** [comment]
@gioman I agree with @asgharibraheembaloch , this is clearly a bug. Using the d&d does not necessarily mean using containers. I have used it plenty of times just to easily hide fields when digitizing.
I cannot think of a reason why not to have a proper layout on the default d&d form. It would also fix the vertical spacing between entries!

**Comment by asgharibraheembaloch** [comment]
thanks @gioman for changing issues title and label
